tsfresh/feature_extraction/extraction.py

```diff
@@ -185,36 +186,39 @@ def _extract_features_parallel_per_sample(kind_to_df_map, settings, column_id, c
     :return: The (maybe imputed) DataFrame containing extracted features.
     :rtype: pandas.DataFrame
     """
+
     partial_extract_features_for_one_time_series = partial(_extract_features_for_one_time_series,
                                                            column_id=column_id,
                                                            column_value=column_value,
                                                            settings=settings)
     pool = Pool(settings.n_processes)
 
-    # Submit map jobs per kind per sample
-    results_fifo = Queue()
-    for kind, df_kind in kind_to_df_map.items():
-        df_grouped_by_id = df_kind.groupby(column_id)
-        results_fifo.put(
-            pool.map_async(
-                partial_extract_features_for_one_time_series,
-                [(kind, df_group) for _, df_group in df_grouped_by_id],
-                chunksize=settings.chunksize
-            )
-        )
+    # Submit map jobs per kind per sample in a generator object
+    kind_and_df_group_list = ((kind, df_group)
```

Comment:
That decision was made when we introduced the multiprocessing Pool:
`extracted_features = pool.map(partial_extract_features_for_one_time_series, kind_to_df_map.items(),
                                  chunksize=settings.chunksize)`